file_processing/tools/transcription_decorator.py

```diff
@@ -22,8 +22,10 @@ def extract_text_with_whisper(self) -> str:
             list: The extracted text, language
         """
         try:
-            model = whisper.load_model('base')
-            text = model.transcribe(str(self._processor.file_path), fp16=False)
+            has_gpu = torch.cuda.is_available()
+            device = 'cuda' if has_gpu else None
+            model = whisper.load_model('base', device=device)
+            text = model.transcribe(str(self._processor.file_path), fp16=has_gpu)
```

Comment:
Have we been able to confirm that the GPU is being used? Any noticable difference in performance?

Reply:
I don't have any device with cuda support so I'm unable to test. I'll try implementing support for `mps` (for Mac) using the same code and see if it works

Reply:
has_gpu came back false on my machine with CUDA

Reply:
I suspect that the issue is that the requirements file brings in torch without CUDA. I think we need to tweak the install of torch to have the option to bring in the CUDA version if available.

Reply:
`whisper` doesn't have `mps` compatibility. Would anyone else be able to test it?

```py
dtype = torch.float16 if decode_options.get("fp16", True) else torch.float32
if model.device == torch.device("cpu"):
    if torch.cuda.is_available():
        warnings.warn("Performing inference on CPU when CUDA is available")
    if dtype == torch.float16:
        warnings.warn("FP16 is not supported on CPU; using FP32 instead")
        dtype = torch.float32

...

parser.add_argument("--device", default="cuda" if torch.cuda.is_available() else "cpu", help="device to use for PyTorch inference")
```

Reply:
I was able to test it on Google Collab and CUDA is marginally faster for the transcription of a ~12 second audio file

![image](https://github.com/hc-sc-ocdo-bdpd/file-processing-tools/assets/155466242/63f2c612-53f1-4b38-9f95-031e26e5fc75)